src/main/java/co/cask/spanner/source/PartitionInputSplit.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.spanner.source;
+
+import co.cask.spanner.SpannerConstants;
+import co.cask.spanner.common.SpannerUtil;
+import com.google.cloud.spanner.BatchClient;
+import com.google.cloud.spanner.BatchReadOnlyTransaction;
+import com.google.cloud.spanner.BatchTransactionId;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Partition;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.MapWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Spanner partition input split,
+ * performs serialization and deserialization of the {@link Configuration}, {@link Partition} and
+ * {@link BatchTransactionId} received from {@link SpannerInputFormat}.
+ *
+ * Executes the spanner query for the {@link Partition}
+ * and makes the result set available to {@link SpannerRecordReader}
+ */
+public class PartitionInputSplit extends InputSplit implements Writable {
+  private static final Logger LOG = LoggerFactory.getLogger(PartitionInputSplit.class);
+  private Partition partition;
+  private Configuration configuration;
+  private BatchTransactionId batchTransactionId;
+
+  public PartitionInputSplit() {
+
+  }
+
+  public PartitionInputSplit(Configuration configuration, Partition partition,
+                             BatchTransactionId batchTransactionId) throws Exception {
+    this.configuration = configuration;
+    this.partition = partition;
+    this.batchTransactionId = batchTransactionId;
+  }
+
+  @Override
+  public long getLength() throws IOException, InterruptedException {
+    // TODO estimate length
```

Comment:
if this is an actual todo, it should have a jira. If we no plans on doing this, remove the comment.